pgp/keysource.go

```diff
@@ -64,6 +108,26 @@ func (key *MasterKey) Encrypt(dataKey []byte) error {
 	return nil
 }
 
+// Encrypt encrypts the data key with the PGP key with the same fingerprint as the MasterKey. It looks for PGP public keys in $PGPHOME/pubring.gpg.
+func (key *MasterKey) Encrypt(dataKey []byte) error {
+	log.Printf("Attempting encryption of GPG MasterKey with fingerprint %s", key.Fingerprint)
+	openpgpErr := key.encryptWithCryptoOpenPGP(dataKey)
+	if openpgpErr == nil {
+		log.Printf("Encryption of GPG MasterKey with fingerprint %s succeeded", key.Fingerprint)
+		return nil
+	}
+	log.Print("Encryption with golang's openpgp package failed, falling back to the GPG binary")
+	binaryErr := key.encryptWithGPGBinary(dataKey)
+	if binaryErr == nil {
+		log.Printf("Encryption of GPG MasterKey with fingerprint %s succeeded", key.Fingerprint)
+		return nil
+	}
+	log.Printf("Encryption of GPG MasterKey with fingerprint %s failed", key.Fingerprint)
+	return fmt.Errorf(`could not encrypt data key with PGP key.
+	\tgolang.org/x/crypto/openpgp error: %s
+	\tGPG binary error: %s`, openpgpErr, binaryErr)
```

Comment:
should we return multiple error strings?

Reply:
I suppose it'd be even better to define our own error type containing the two possible errors with an `Error() string` function.